### PR TITLE
fix: MusixMatch url sanitize

### PR DESCRIPTION
**Pull request by aymanbagabas**
* Sanitize MusixMatch URL to match on more songs
* Fix SongLyrics out of bound error

### CODE REVIEW

File: musixmatch/fetch.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ func New() *MusixMatch {
 func (*MusixMatch) Fetch(artist, song string) string {
 	url := "https://www.musixmatch.com/lyrics/" + formatURL(artist) + "/" + formatURL(song)
 
+	log.Printf("src: %s", url)
```
Comment (rhnvrm) [inline musixmatch/fetch.go]: Please remove the debug log.

File: musixmatch/helper.go [hunk 2]
```diff
@@ -1,7 +1,15 @@
 package musixmatch
 
-import "strings"
+import (
+	"regexp"
+	"strings"
+)
+
+var urlRegexp = regexp.MustCompile(`[^a-zA-Z0-9 ]`)
```
Comment (rhnvrm) [inline musixmatch/helper.go]: Maybe we can format it as 
```
var (
  urlRegexp = ...
  whitespaceRegexp = ...
)

**Review by rhnvrm — CHANGES_REQUESTED** [review]
LGTM, few minor suggestions

File: musixmatch/helper.go (hunk 2 above)
Comment (aymanbagabas) [reply musixmatch/helper.go]: done

**Review by rhnvrm — APPROVED** [review]

**Pull request merged**